bundles/org.openhab.binding.enphase/src/main/java/org/openhab/binding/enphase/internal/handler/EnvoyBridgeHandler.java

```diff
@@ -0,0 +1,410 @@
+            return;
+        }
+        if (configuration.hostname.isEmpty()) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.CONFIGURATION_PENDING,
+                    "Waiting to retrieve ip address of the envoy gateway. Can take up to a minute.");
+        } else {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        connector.setConfiguration(configuration);
+        consumptionSupported = FeatureStatus.UNKNOWN;
+        jsonSupported = FeatureStatus.UNKNOWN;
+        invertersCache = new ExpiringCache<>(Duration.of(configuration.refresh, ChronoUnit.MINUTES),
+                this::refreshInverters);
+        devicesCache = new ExpiringCache<>(Duration.of(configuration.refresh, ChronoUnit.MINUTES),
+                this::refreshDevices);
+        updataDataFuture = scheduler.scheduleWithFixedDelay(this::updateData, 0, configuration.refresh,
+                TimeUnit.MINUTES);
+    }
+
+    /**
+     * Method called by the ExpiringCache when no inverter data is present to get the data from the Envoy gateway.
+     * When there are connection problems it will start a scheduled job to try to reconnect to the
+     *
+     * @return the inverter data from the Envoy gateway or null if no data is available.
+     */
+    private @Nullable Map<String, @Nullable InverterDTO> refreshInverters() {
+        try {
+            return connector.getInverters().stream()
+                    .collect(Collectors.toMap(InverterDTO::getSerialNumber, Function.identity()));
+        } catch (final EnvoyNoHostnameException e) {
+            // ignore hostname exception here. It's already handled by others.
+        } catch (final EnvoyConnectionException e) {
+            logger.trace("refreshInverters connection problem", e);
+        }
+        return null;
+    }
+
+    private @Nullable Map<String, @Nullable DeviceDTO> refreshDevices() {
+        try {
+            if (jsonSupported != FeatureStatus.UNSUPPORTED) {
+                final Map<String, @Nullable DeviceDTO> devicesData = connector.getInventoryJson().stream()
+                        .flatMap(inv -> Stream.of(inv.devices).map(d -> {
+                            d.type = inv.type;
+                            return d;
+                        })).collect(Collectors.toMap(DeviceDTO::getSerialNumber, Function.identity()));
+
+                jsonSupported = FeatureStatus.SUPPORTED;
+                return devicesData;
+            }
+        } catch (final EnvoyNoHostnameException e) {
+            // ignore hostname exception here. It's already handled by others.
+        } catch (final EnvoyConnectionException e) {
+            if (jsonSupported == FeatureStatus.UNKNOWN) {
+                logger.info(
+                        "This Ephase Envoy device ({}) doesn't seem to support json data. So not all channels are set.",
+                        getThing().getUID());
```

Comment:
Logging to info should be used sparsely e.g. a newly started component or a user file that has been loaded. This could be debug. Please check all.